src/logic/player.cc

```diff
@@ -2293,4 +2306,40 @@ void Player::write_statistics(FileWrite& fw) const {
 	}
 }
 
+std::pair<std::set<std::string>, std::set<std::string>> read_custom_warehouse_ship_names() {
+	std::pair<std::set<std::string>, std::set<std::string>> result;
+
+	if (FileRead fr; fr.try_open(*g_fs, kCustomShipNamesFile)) {
+		for (;;) {
+			const char* line = fr.read_line();
+			if (line == nullptr) {
+				break;
+			}
+
+			std::string name(line);
+			trim(name);
+			if (!name.empty()) {
+				result.first.insert(name);
```

Comment:
Don't the names need richtext escaping as well? And possibly removal of control chars too?

Reply:
Good catch, thanks.

Control chars are already taken care of by the font tag formatting:

![grafik](https://github.com/widelands/widelands/assets/48293446/e8553024-3e20-444b-b8da-bccb9c167261)